src/main/java/org/testng/internal/ScriptMethodSelector.java

```diff
@@ -0,0 +1,54 @@
+package org.testng.internal;
+
+import org.testng.ITestNGMethod;
+import org.testng.TestNGException;
+import org.testng.collections.Maps;
+
+import javax.script.ScriptContext;
+import javax.script.ScriptEngine;
+import javax.script.ScriptException;
+import java.lang.reflect.Method;
+import java.util.Map;
+
+public class ScriptMethodSelector {
+
+  private final ScriptEngine engine;
+
+  public ScriptMethodSelector(ScriptEngine engine) {
+    this.engine = engine;
+  }
+
+  public boolean includeMethodFromExpression(String expression, ITestNGMethod tm) {
+    Map<String, String> groups = Maps.newHashMap();
+    for (String group : tm.getGroups()) {
+      groups.put(group, group);
+    }
+    try {
+      setContext(engine, tm.getConstructorOrMethod().getMethod(), groups, tm);
+      Object evalResult = engine.eval(expression.trim());
+      if (evalResult == null) {
+        String msg = String.format("The " + engine.getFactory().getLanguageName() + " expression [%s] evaluated to null.", expression);
+        throw new TestNGException(msg);
+      }
+      return (Boolean) evalResult;
+    } catch (ScriptException e) {
+      throw new TestNGException(e);
+    } finally {
+      resetContext(engine);
+    }
+  }
+
+  private static void setContext(ScriptEngine engine, Method method, Map<String, String> groups, ITestNGMethod tm) {
```

Comment:
The parameter `Method method` can be removed since it can be internally extracted out of `tm` object. So one less parameter for the method.

Reply:
Right! It was the original code and I hesitated to change it ;)